idx/memory/memory_find_test.go

```diff
@@ -391,6 +392,260 @@ func TestTagKeysWithFromAndFilter(t *testing.T) {
 	queryAndCompareTagKeys(t, "di", 1000000, nil)
 }
 
+func TestTagSorting(t *testing.T) {
+	index := New()
+	index.Init()
+
+	md1 := &schema.MetricData{
+		Name:     "name1",
+		Metric:   "name1",
+		Tags:     []string{},
+		Interval: 10,
+		OrgId:    1,
+		Time:     int64(123),
+	}
+	md1.SetId()
+
+	// set out of order tags after SetId (because that would sort it)
```

Comment:
it's not clear what we're actually trying to test here. we assume SetId correctly sorts, so we unsort again. why?

Reply:
because theoretically even though `SetId()` sorts the tags, somebody who modifies their crng could still send MDs with out-of-order tags, so I sort the incoming tags just in case.